src/lamp_py/aws/s3.py

```diff
@@ -321,6 +321,37 @@ def file_list_from_s3_with_details(
         return []
 
 
+def get_last_modified_object(
+    bucket_name: str, file_prefix: str, version: Optional[str] = None
+) -> Optional[Dict]:
+    """
+    For a given bucket, find the last modified object that matches a prefix. If
+    a version is passed, only return the newest object matching this version.
+    """
+    files = file_list_from_s3_with_details(
+        bucket_name=bucket_name, file_prefix=file_prefix
+    )
+
+    # sort the objects by last modified
+    files.sort(key=lambda o: o["last_modified"], reverse=True)
+
+    # if there are no objects, return None
+    if len(files) == 0:
+        return None
+
+    # if there is no version, the return the first object
+    if version is None:
+        return files[0]
+
+    # if there is a version, return the first object that matches the version
+    for file in files:
+        if version_check(file["s3_obj_path"], version):
+            return file
+
+    # if we were unable to match an object, return None
+    return None
+
+
```

Comment:
i updated this function to only return the file that also has a version match.

Reply:
i ran this against mbta-ctd-dataplatform-dev-springboard/lamp/RT_VEHICLE_POSITIONS, which has 4054 files with a version match and it took 6 and half minutes. if i drop the version match it takes 1.2 seconds. i get a similar runtime if i do a vesion match against the empty string (the version info of all these files).